src/main/java/cz/cvut/fel/aic/simod/ridesharing/insertionheuristic/InsertionHeuristicSolver.java

```diff
@@ -429,17 +429,17 @@ private DriverPlan insertIntoPlan(
 			}
 
 			// travel time increment
-			int travelTime;
+			long travelTime;
 			if (previousTask instanceof PlanActionCurrentPosition) {
-				travelTime = (int) travelTimeProvider.getTravelTime(vehicle, newTask.getPosition()) / 1000;
+				travelTime = travelTimeProvider.getTravelTime(vehicle, newTask.getPosition());
 			} else {
-				travelTime = (int) travelTimeProvider.getTravelTime(vehicle, previousTask.getPosition(),
+				travelTime = travelTimeProvider.getTravelTime(vehicle, previousTask.getPosition(),
 					newTask.getPosition()
-				) / 1000;
+				);
 			}
 			newPlanTravelTime += travelTime;
-			currentTaskTimeInSeconds += travelTime;
-			timeWithoutPause += travelTime;
+			currentTaskTimeInSeconds += travelTime / 1000;
+			timeWithoutPause += travelTime / 1000;
```

Comment:
I think we should also use the same approach for these variables to prevent subtle errors with constraint violations